src/common/w25x_communication.cpp

```diff
@@ -5,6 +5,8 @@
 #include "cmsis_os.h"
 #include "stm32f4xx_hal.h"
 #include "hwio_pindef.h"
+#define F_CPU SystemCoreClock
+#include "../../lib/Marlin/Marlin/src/HAL/shared/Delay.h"
```

Comment:
[DRracer](https://github.com/DRracer) [yesterday](https://github.com/prusa3d/Prusa-Firmware-Buddy/pull/2036#discussion_r854176409)

I don't like having dependency on Marlin in dump.c, that's completely unrelated. Don't we have a DELAY_NS in some other form? AFAIK it all boils down to tick_ns() or a similar call, which is our code.

Reply:
I agree with @DRracer here. We definitely don't want to have this code dependent on Marlin here.

We should just add `delay_ns` (and other) functions to our `timing.h`.

Reply:
fixed